src/state/Autocomplete.ts

```diff
@@ -0,0 +1,113 @@
+    ]
+  }
+
+  private async getMatchingDepartements(term: NormalizedSearch): Promise<Departement[]> {
+    const departements = await this.getDepartements()
+    return departements.filter((departement) => {
+      return departement.code_departement.startsWith(term)
+          || this.normalize(departement.nom_departement).includes(term)
+    })
+  }
+
+  private async getMatchingCommunes(term: NormalizedSearch): Promise<Commune[]> {
+    const prefixMatch = await this.getLongestPrefixMatch(term)
+    if (!prefixMatch) {
+      return []
+    }
+    const communesWithPrefix = await this.getAutocompleteOptions(prefixMatch)
+    return communesWithPrefix.filter((commune) => {
+      return commune.codePostal.includes(term)
+          || this.normalize(commune.nom).includes(term)
+    })
+  }
+
+  private mapAutocommpleteToCommune(option: CommuneAutocomplete): Commune {
+    const [ longitude, latitude ] =  option.g.split(',').map(Number)
+    return {
+      nom: option.n,
+      code: option.c,
+      codePostal: option.z,
+      codeDepartement: option.d,
+      latitude, longitude
+    }
+  }
+
+  private async getLongestPrefixMatch(term: NormalizedSearch): Promise<NormalizedSearch | undefined> {
+    const prefixes = await this.getAutocompletePrefixes()
+    let longestMatch = term
+    for (let size = term.length; size > 0; --size) {
+      const subPrefix = term.substring(0, size)
+      if (prefixes.has(subPrefix)) {
+        return subPrefix as NormalizedSearch
+      }
+    }
+    return undefined
+  }
+
+  @Memoize()
+  private async getDepartements (): Promise<Departement[]> {
+    return this.getDepartementsDisponibles()
+  }
+
+  @Memoize()
+  private async getAutocompletePrefixes(): Promise<Set<string>> {
+    const response = await window.fetch(`${this.webBaseUrl}autocompletes.json`)
+    const prefixes = (await response.json()) as string[]
+    return new Set<string>(prefixes)
+  }
+
+  @Memoize()
+  private async getAutocompleteOptions(name: NormalizedSearch): Promise<Commune[]> {
+    const response = await window.fetch(`${this.webBaseUrl}autocomplete-cache/${name}.json`)
+    const { communes } = await response.json() as { communes: CommuneAutocomplete[] }
+    return communes.map(this.mapAutocommpleteToCommune)
+  }
+
+
+  private normalize (term: string): NormalizedSearch {
+    // /!\ important note : this is important to have the same implementation of toFullTextSearchableString()
+    // function here, than the one used in communes-import.js tooling
+    // FIXME use actual same code instead of warning, its in ../../tools/communes-import.js
+    return term.toLowerCase()
+      .replace(/[-\s']/gi, "_")
+      .replace(/[èéëêêéè]/gi, "e")
+      .replace(/[áàâäãåâà]/gi, "a")
+      .replace(/[çç]/gi, "c")
+      .replace(/[íìîï]/gi, "i")
+      .replace(/[ñ]/gi, "n")
+      .replace(/[óòôöõô]/gi, "o")
+      .replace(/[úùûüûù]/gi, "u")
+      .replace(/[œ]/gi, "oe") as NormalizedSearch
```

Comment:
Hello, il existe une méthode [String.prototype.normalize()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize) en natif qui permet de sortir les accents et de les remplacer d'un coup.
```suggestion
    return term.toLowerCase()
      .normalize("NFD")
      .replace(/[\u0300-\u036f]/g, "")
      .replace(/[-\s']/gi, "_")
      .replace(/[œ]/gi, "oe") as NormalizedSearch
```

Reply:
effectivement =)
faut voir si ça génère exactement les mêmes string. SI non, il faudra regénérer toutes les ressources d'autocomplétion avec le même algo. si c'est le cas je préfère le faire dans une PR dédiée à ça.